static_website/dispatch.ml

```diff
@@ -1,53 +1,50 @@
 open Lwt.Infix
-open V1_LWT
 
-module Main (C:CONSOLE) (FS:KV_RO) (S:Cohttp_lwt.Server) = struct
+let server_src = Logs.Src.create "server" ~doc:"HTTP server"
+module Server_log = (val Logs.src_log server_src : Logs.LOG)
 
-  let start c fs http =
+module Main (Clock:V1.CLOCK) (FS:V1_LWT.KV_RO) (S:Cohttp_lwt.Server) = struct
+  module Logs_reporter = Mirage_logs.Make(Clock)
 
-    let read_fs name =
-      FS.size fs name >>= function
+  let read_fs fs name =
+    FS.size fs name >>= function
+    | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
+    | `Ok size ->
+      FS.read fs name 0 (Int64.to_int size) >>= function
       | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
-      | `Ok size ->
-        FS.read fs name 0 (Int64.to_int size) >>= function
-        | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
-        | `Ok bufs -> Lwt.return (Cstruct.copyv bufs)
-    in
+      | `Ok bufs -> Lwt.return (Cstruct.copyv bufs)
 
-    (* Split a URI into a list of path segments *)
-    let split_path uri =
-      let path = Uri.path uri in
-      let rec aux = function
-        | [] | [""] -> []
-        | hd::tl -> hd :: aux tl
-      in
-      List.filter (fun e -> e <> "")
-        (aux (Re_str.(split_delim (regexp_string "/") path)))
-    in
+  let rec dispatcher fs uri =
```

Comment:
not sure if this preserves the semantics of before... then not sure whether it needs to preserve the semantics from earlier ;)